frontend/src/pages/auth/signin/SignIn.tsx

```diff
@@ -39,12 +39,14 @@ const SignIn: React.FC = () => {
     }
 
     const handleSignIn = async () => {
-        const token = await logInService(authCredentials);
-        const decodedToken = decodeToken(token);
-        const user = await getUserService(decodedToken.userId, decodedToken.userType);
-        const loggedUser = createLoggedUser(token, user);
-        if (isValid && loggedUser) {
-            dispatch(signIn(loggedUser));
+        if (isValid) {
```

Comment:
reduce nesting, by inverting condition.
```suggestion
        if (!isValid) return;
```